cmd/restic/cmd_copy.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/restic/restic/internal/debug"
+	"github.com/restic/restic/internal/errors"
+	"github.com/restic/restic/internal/restic"
+
+	"github.com/spf13/cobra"
+)
+
+var cmdCopy = &cobra.Command{
+	Use:   "copy [flags] [snapshotID ...]",
+	Short: "Copy snapshots from one repository to another",
+	Long: `
+The "copy" command copies one or more snapshots from one repository to another
+repository. Note that this will have to read(download) and write(upload) the
+entire snapshot(s) due to the different encryption keys on the source and
+destination.
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runCopy(copyOptions, globalOptions, args)
+	},
+}
+
+// CopyOptions bundles all options for the copy command.
+type CopyOptions struct {
+	Repo            string
+	PasswordFile    string
+	PasswordCommand string
+	KeyHint         string
+	Password        string
+	Hosts           []string
+	Tags            restic.TagLists
+	Paths           []string
+}
+
+var copyOptions CopyOptions
+
+func init() {
+	cmdRoot.AddCommand(cmdCopy)
+
+	f := cmdCopy.Flags()
+	f.StringVarP(&copyOptions.Repo, "repo2", "", os.Getenv("RESTIC_REPOSITORY2"), "repository to backup to or restore from (default: $RESTIC_REPOSITORY2)")
+	f.StringVarP(&copyOptions.PasswordFile, "password-file2", "", os.Getenv("RESTIC_PASSWORD_FILE2"), "read the destination repository password from a file (default: $RESTIC_PASSWORD_FILE2)")
+	f.StringVarP(&copyOptions.KeyHint, "key-hint2", "", os.Getenv("RESTIC_KEY_HINT2"), "key ID of key to try decrypting the destination repository first (default: $RESTIC_KEY_HINT2)")
+	f.StringVarP(&copyOptions.PasswordCommand, "password-command2", "", os.Getenv("RESTIC_PASSWORD_COMMAND2"), "specify a shell command to obtain a password for the destination repository (default: $RESTIC_PASSWORD_COMMAND2)")
+
+	f.StringArrayVarP(&copyOptions.Hosts, "host", "H", nil, "only consider snapshots for this `host`, when no snapshot ID is given (can be specified multiple times)")
+	f.Var(&copyOptions.Tags, "tag", "only consider snapshots which include this `taglist`, when no snapshot-ID is given")
+	f.StringArrayVar(&copyOptions.Paths, "path", nil, "only consider snapshots which include this (absolute) `path`, when no snapshot-ID is given")
```

Comment:
Just a proposal: Why not add an option to explicitely specify one or more snapshots by ID? (see `cmd_forget.go`)